core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/AbstractFileBatchSource.java

```diff
@@ -219,7 +220,7 @@ public void onRunFinish(boolean succeeded, BatchSourceContext context) {
    */
   @javax.ws.rs.Path("getSchema")
   public Schema getSchema(SchemaRequest request, EndpointPluginContext pluginContext) {
-    return PathTrackingInputFormat.getOutputSchema(request.pathField);
+    return PathTrackingInputFormat.getTextOutputSchema(request.pathField);
```

Comment:
this gets called when the user hits the 'get schema' button. It should not return the text schema when the source is not configured to use text.

However, now that the user is responsible for setting a schema, we should just remove this method and corresponding widget entirely, since it does not make sense anymore.